python/ray/serve/_private/deployment_state.py

```diff
@@ -2112,6 +2127,13 @@ def check_and_update_replicas(self):
                         "application": self.app_name,
                     },
                 )
+                # if replica version is the same as the target version,
+                # we update the docs path
+                if (
+                    replica.version == self._target_state.version
+                    and replica.docs_path is not None
+                ):
+                    self._docs_path = replica.docs_path
```

Comment:
synced offline, if possible it seems more intuitive to set the docs path upon replica startup